function/expression.go

```diff
@@ -19,18 +18,17 @@ import (
 // * Contains current timerange being queried for - this can be
 // changed by say, application of time shift function.
 type EvaluationContext struct {
-	TimeseriesStorageAPI      api.TimeseriesStorageAPI // Backend to fetch data from
-	MetricMetadataAPI         api.MetricMetadataAPI    // Api to obtain metadata from
-	Timerange                 api.Timerange            // Timerange to fetch data from
-	SampleMethod              api.SampleMethod         // SampleMethod to use when up/downsampling to match the requested resolution
-	Predicate                 api.Predicate            // Predicate to apply to TagSets prior to fetching
-	FetchLimit                FetchCounter             // A limit on the number of fetches which may be performed
-	Cancellable               api.Cancellable
-	Registry                  Registry
-	Profiler                  *inspect.Profiler // A profiler pointer
-	OptimizationConfiguration *optimize.OptimizationConfiguration
```

Comment:
The real change is the removal of the `OptimizationConfiguration` field (the others just have new indentation).

Now, OptimizationConfigurations manipulate the `Context` directly, so there's no need for it to it to be stored or accessed during evaluation